### PR TITLE
Movie with Hibernate added

### PR DESCRIPTION
**Pull request by SetMyDream**

### CODE REVIEW

**Review by Vosker — COMMENTED** [review]
Good job

File: pom.xml [hunk 1]
```diff
@@ -38,6 +34,16 @@
...
+            <version>5.4.21.Final</version>
+        </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-java8</artifactId>
+            <version>5.4.5.Final</version>
+        </dependency>
     </dependencies>
     <build>
         <plugins>
```
Comment (KseniiaMakarova) [inline pom.xml:49]: let's remove war plugin

File: src/main/java/com/dev/cinema/models/Movie.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+@Data
+@NoArgsConstructor
+@Entity
+@Table(name = "movies")
+public class Movie {
+    @Id
+    @Column(name = "movie_id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+        private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/models/Movie.java]: intendation level

File: src/main/java/com/dev/cinema/models/Movie.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        this.title = title;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/models/Movie.java:46]: what do you see in your console when you run the main class?

File: src/main/java/com/dev/cinema/service/impl/MovieServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.models.Movie;
+import com.dev.cinema.service.MovieService;
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+    @Inject
+    MovieDao movieDao;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/impl/MovieServiceImpl.java]: let's add access modifier

File: src/main/resources/hibernate.cfg.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version='1.0' encoding='utf-8'?>
+<!DOCTYPE hibernate-configuration PUBLIC
+        "-//Hibernate/Hibernate Configuration DTD//EN"
+        "http://www.hibernate.org/dtd/hibernate-configuration-3.0.dtd">
+
+<hibernate-configuration>
+    <session-factory>
+        <property name="dialect">org.hibernate.dialect.MySQL5Dialect</property>
+        <property name="hibernate.connection.url">jdbc:mysql://localhost:3306/cinema?serverTimezone=UTC</property>
+        <property name="hibernate.connection.driver_class">com.mysql.jdbc.Driver</property>
```
Comment (KseniiaMakarova) [inline src/main/resources/hibernate.cfg.xml]:
```suggestion
        <property name="hibernate.connection.driver_class">com.mysql.cj.jdbc.Driver</property>
```

File: src/main/resources/init_db.sql [hunk 6]
```diff
@@ -0,0 +1,8 @@
+CREATE SCHEMA `cinema` DEFAULT CHARACTER SET utf8 ;
+
+CREATE TABLE `cinema`.`movies` (
+                                   `movie_id` INT NOT NULL AUTO_INCREMENT,
+                                   `title` VARCHAR(256) NOT NULL,
+                                   `description` VARCHAR(4096) NOT NULL,
+                                   `deleted` TINYINT NOT NULL DEFAULT '0',
+                                   PRIMARY KEY (`movie_id`));
```
Comment (KseniiaMakarova) [inline src/main/resources/init_db.sql]: let's remove this - our tables will be created by hibernate

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/models/Movie.java (hunk 3 above)
Comment (SetMyDream) [reply src/main/java/com/dev/cinema/models/Movie.java:46]: was description=null
but i`ll change to "No description provided"

Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/models/Movie.java:46]: I see that you use Lombok
no changes are needed, sorry

File: src/main/java/com/dev/cinema/models/Movie.java [hunk 7]
```diff
@@ -7,20 +7,24 @@
...
-        private Long id;
+    private Long id;
     private String title;
     private String description;
 
+    public Movie() {
+        this.description = "No description provided";
+        this.title = "No title provided";
+        this.id = -1L;
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/models/Movie.java]: let's remove this and return to usage of @NoArgsConstructor

and can we remove war-plugin from pom.xml please?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**